spec/requests/memberships_controller_spec.rb

```diff
@@ -285,4 +285,25 @@
       end
     end
   end
+
+  describe '#withdraw' do
+    include_context 'current_user is a member'
+    context 'when membership has DEFAULT post' do
+      let(:membership) { create(:membership, :newbie) }
+      it 'withdraws the membership' do
+        post "/groups/#{membership.group.id}/memberships/#{membership.id}/withdraw", xhr: true
+
+        expect(response).to redirect_to group_path(group)
+      end
```

Comment:
I feel a bit inconsistency between the description of the spec and the expectation. In case of `withdraws the membership` I would expect that there is a check whether the membership is deleted.

```suggestion
      it 'withdraws the membership' do
        post "/groups/#{membership.group.id}/memberships/#{membership.id}/withdraw", xhr: true

        expect(response).to redirect_to group_path(group)
        # Probably there is a nicer way to check this
        expect(Membership.exists?(id: membership.id)).to be false
      end
```